ZOOKEEPER-2801: address spelling errors/typos

tmancill opened the pull request.

bensherman commented:
Awesome, thanks for doing this!

afine commented on src/contrib/loggraph/web/org/apache/zookeeper/graph/resources/raphael.js:
```diff
@@ -3018,9 +3018,9 @@ Raphael = (function () {
-                        var pathes = path2curve(from[attr], to[attr]);
-                        from[attr] = pathes[0];
-                        var toPath = pathes[1];
+                        var paths = path2curve(from[attr], to[attr]);
```
lets not make unnecessary code changes

afine commented on src/java/main/org/apache/zookeeper/server/WatchManager.java:
```diff
@@ -94,7 +94,7 @@ synchronized void removeWatcher(Watcher watcher) {
     }
 
-    Set<Watcher> triggerWatch(String path, EventType type, Set<Watcher> supress) {
+    Set<Watcher> triggerWatch(String path, EventType type, Set<Watcher> suppress) {
```
I think we should avoid code changes here.

tmancill commented:
@afine Thank you for the careful review, and that's a good point about avoiding code changes.   In b152ed3 I revert the changes you mention.  I reviewed the patch again and believe those were the only two instances where I modified code.

hanm commented:
Nice pull request.

Please don't directly modify the document artifacts (html files, etc). The way the document is updated is by modifying the source of the docs, located at src/docs/src/documentation/content/xdocs. After modifying the source, please verify the generated document is correct locally by using apache forrest https://forrest.apache.org/. After verification please submit the document source change only - the compiled document artifacts (html files, etc) don't need to be submitted, because we will regenerate document in every release. Please check the commit history of https://github.com/apache/zookeeper/tree/master/src/docs/src/documentation/content/xdocs to get a concrete idea of how to make doc changes, it should be pretty straightforward.

It would be also good to fix the similar typos in branch-3.5 and branch-3.4, which this PR does not directly apply with many merge conflicts.

hanm commented:
@tmancill Also I am curious what tools you use to catch the spelling errors. I think have such tool be part of commit workflow, or daily build would be helpful.

tmancill commented:
@hanm Thank you for the information regarding the documentation tool-chain.  I will rework the PR to only modify the document sources.  (Although it will probably be early next week.)

The tool used to find the typos is [spellintian](https://manpages.debian.org/unstable/lintian/spellintian.1.en.html), which is part of Debian's [lintian](https://tracker.debian.org/pkg/lintian) package.  It's pretty good at catching some common typos, although certain types of code documentation trigger false positives for repeated words when the source is _word punctuation word_.  For this PR, I fed the spellintian output into some `awk` to produce `sed` commands of proposed corrections.  Then I removed the sed commands for the false positives and other edge cases (still a manual step) and ran the script against the source tree.  So it's conceivable that spellintian could be worked into a commit workflow, although given all of the moving parts, maybe it fits better as part of a pre-release checklist.

Regarding branch-3.4, https://github.com/apache/zookeeper/pull/174 should have addressed most of the issues in that branch.  `lintian` didn't notice any typos when I packaged 3.4.10 for Debian.   

And for branch-3.5, this patch will apply after dropping a couple of the hunks.  Should I create a separate JIRA and submit a separate PR (once I get this one correct) ?

hanm commented:
>> although given all of the moving parts, maybe it fits better as part of a pre-release checklist.

Sounds good to me. 

>> Should I create a separate JIRA and submit a separate PR (once I get this one correct)

Yes please create a separate PR for branch-3.5.

>> Although it will probably be early next week.

Thank you, no rush.
